### ISSUE TITLE
Import log (.chls) and config (.xml) files from Charles

### ISSUE THREAD

**Issue by NghiaTranUIT** [ProxymanApp/Proxyman#483]
### Description
Some users request that Proxyman would be able to import .chls file from Charles Proxy, so they can migrate their work to Proxyman easily.

We should support it.

### Acceptance Criteria 
- Import .chls log file
- Import .xml file for all Charles Setting (Map Local, Remote, Breakpoint, ...)

**Comment by dougmcbride** [comment]
It would be even better if logs could be saved in the .chls format as well. This way people who work in Charles shops could switch over to Proxyman without forcing the whole team to move at once.

Would this be a separate ticket, or part of this one?

**Comment by NghiaTranUIT** [comment]
Good point @dougmcbride  👍 I will update the ticket's description to support export files back to Charles since the Serialization and Deserialization are mostly the same 😄

**Comment by NghiaTranUIT** [comment]
Meanwhile, @dougmcbride if you need to import/export flow data between Charles and Proxyman, you can export as **HAR**, which is widely supported by other apps like Charles and Google Chrome.

https://docs.proxyman.io/basic-features/import-export

**Comment by grschom** [comment]
A good workaround until ProxyMan support opening `.chls` files natively 👍 

FYI, I noticed Charles has a [command line interface](https://www.charlesproxy.com/documentation/tools/command-line-tools/) that I successfully used to convert a `.chls` to `.har` and open via proxyman.

**Comment by NghiaTranUIT** [comment]
Thank for the idea. I successfully convert .chls to .har via CLI.

In order to directly import `.chls` to Proxyman, Charles has to be installed in the user machine, so Proxyman will call Charles to convert to `.har` then importing to Proxyman.

I will support this feature in the next release 👍

**Comment by NghiaTranUIT** [comment]
Hey @grschom and @dougmcbride, let check out this Beta build https://proxyman.s3.us-east-2.amazonaws.com/beta/Proxyman_1.22.0_Import_Charles_Proxy_File.dmg
I gonna release today or tmr

From now, you can:
- Open directly `.chls` file or drag the file to Proxyman to open
- It requires that Charles Proxy app is installed
- Or you can open the Converter Window in File Menu -> Convert

Thank you 💪

![2020-05-03 10 38 48](https://user-images.githubusercontent.com/5878421/80907793-ed31a380-8d43-11ea-89ec-f1c569d9ef4c.gif)

**Comment by NghiaTranUIT** [comment]
Just release Proxyman 1.23.0: https://github.com/ProxymanApp/Proxyman/releases/tag/1.23.0

From now, we can open Charles Log directly 👍 @grschom @dougmcbride

**Comment by grschom** [comment]
awesome! Thanks for implementing this so quickly!

**Comment by benjidea** [comment]
The Charles Map-Local-Export contains only a relative reference to the file. 

An extended import/export for Map-Local including the files would be very useful.
Do we need another Issue for this?

**Comment by NghiaTranUIT** [comment]
Thanks for raising it. It's perfect timing to support this feature @benjidea 👍 

I created a new ticket for it: https://github.com/ProxymanApp/Proxyman/issues/591

**Comment by danimal** [comment]
It would be great to also support the `.chlsj` (Charles JSON) files. I know we can use `.har` and I have been doing that to great success. I sometimes get `.chlsj` files from people that have Charles Proxy for iOS and ProxyMan can't open them. If this needs to be a new Issue I can do that as well.

**Comment by bskendig** [comment]
A way for Proxyman to directly export a `.chls` file hasn't been added yet, right? I'm interested in this for the same reasons @dougmcbride gave above.

**Comment by NghiaTranUIT** [comment]
Sorry for the late reply @bskendig. I missed the notification.
-------------
Charles Proxy doesn't publicize the structure of the `.chls` file, so I don't know how to properly construct them.

However, not sure if you're aware of:

- Proxyman can directly open `.chls`. Document at https://docs.proxyman.io/advanced-features/charles-proxy-converter
- Proxyman can export as a HAR file, then use Charles Proxy to open it.
